### PR TITLE
images: Fixes CreateImage

### PR DESCRIPTION
**Pull request by ldjebran**
create a getImageSetForNewImage func. that
Check for ImageSet existence,
if imageSet does not exist create one,
if it exists and is not linked to any images reuse it,
if it exists and linked to any images return error

Fixes #THEEDGE-2145

## Type of change

What is it?

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [X] Refactor

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `go fmt ./...` to check that my code is properly formatted
- [X] I run `go vet ./...` to check that my code is free of common Go style mistakes
- [X ] I run `make lint` to prints out coding style mistakes and fix them

### CODE REVIEW

File: pkg/services/images.go [hunk 1]
```diff
@@ -108,6 +108,41 @@ func ValidateAllImageReposAreFromAccount(account string, repos []models.ThirdPar
 	return nil
 }
 
+func (s *ImageService) getImageSetForNewImage(account string, image *models.Image) (*models.ImageSet, error) {
```
[the remark this answers is not in the document]
Comment (ldjebran) [reply pkg/services/images.go:111]: anyway nothing special here, 2 return of imageSet, and 2 errors returns.

[the remark this answers is not in the document]
Comment (AaronH88) [reply pkg/services/images.go:111]: Sounds ok to me :smile:

**Comment by AaronH88** [comment]
/lgtm

ok to merge

**Review by AaronH88 — APPROVED** [review]

**Comment by ldjebran** [comment]
@Mergifyio refresh

**Comment by ldjebran** [comment]
@Mergifyio requeue

**Comment by ldjebran** [comment]
@Mergifyio unqueue

**Comment by ldjebran** [comment]
@Mergifyio refresh

**Comment by ldjebran** [comment]
@Mergifyio queue

**Comment by ldjebran** [comment]
@Mergifyio requeue

**Comment by ldjebran** [comment]
@Mergifyio refresh

**Pull request merged**